include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -3023,6 +3023,16 @@ struct OrtApi {
   */
   ORT_API2_STATUS(GetSparseTensorIndices, _In_ const OrtValue* ort_value, enum OrtSparseIndicesFormat indices_format, _Out_ size_t* num_indices, _Outptr_ const void** indices);
 
+  /** \brief Used for custom operators, gets the CUDA stream to use to launch the custom CUDA kernel     
+  *   \see ::OrtCustomOp
+  * \param[context] OrtKernelContext instance
+  * \param[out] Returns pointer to a CUDA stream that can be used to launch the custom CUDA kernel.
+  *             If retrieving the CUDA stream is not relevant (CUDA not enabled in the build, kernel partitioned to
+  *             some other EP), then a nullptr is returned as the output param.
+  *             Do not free or mutate the returned pointer as it refers to internal data owned by the underlying session.
+  *             Only use it for custom kernel launching.
+  */
+  ORT_API2_STATUS(KernelContext_GetCUDAStream, _In_ const OrtKernelContext* context, _Outptr_ void** out);
```

Comment:
Should CUDA (a specific EP) be in the name of the API? The implementation doesn't even check for CUDA. Besides, this API could be used for ROCM EP as well. Should we rename this to KernelContext_GetGPUComputeStream (since it's really the compute stream that we're returning)?